swap_meet/vendor.py

```diff
@@ -1,2 +1,109 @@
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        # try the item exists in the inventory, remove it and return the removed item, else raise error
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+
+    def get_by_id(self, id):
+        if not isinstance(id, int):
+            raise TypeError("id must be an integer")
+        # Iterate through the inventory to find an item with a matching id
+        for item in self.inventory:  
+            # If an item with a matching id is found, return it  
+            if item.id == id:
+                return item
+        # If no matching item is found, return None
+        return None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        # If either the `my_item` is not in this Vendor's inventory or the `their_item` 
+        # is not in the other Vendor's inventory, return False
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        # Remove `my_item` from this Vendor's inventory and add it to the other Vendor's inventory
+        self.remove(my_item)
+        other_vendor.inventory.append(my_item)
+        # Remove `their_item` from the other Vendor's inventory and add it to this Vendor's inventory
+        other_vendor.inventory.remove(their_item)
+        self.add(their_item)
+        # If successful, return True
+        return True
+    
+    # def swap_first_item(self, other_vendor):
+    #     # Check if either the vendor or the friend have empty inventory, and return False if so
+    #     if not self.inventory or not other_vendor.inventory:
+    #         return False
+    #     # Swap the first item from the friend's inventory with the first item from the vendor's inventory
+    #     return self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+    def swap_first_item(self, other_vendor):
+        try:
+            # Check if either the vendor or the friend have empty inventory, and return False if so
+            if not self.inventory or not other_vendor.inventory:
+                return False
+            # Swap the first item from the friend's inventory with the first item from the vendor's inventory
+            return self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+        except IndexError:
+            return False
+
+    def get_by_category(self, category):
+        # Create an empty list to store the matching items
+        items_matching_category = []
+        # Iterate through the inventory to find items with matching category
+        for item in self.inventory:
+            if item.get_category() == category:
+                items_matching_category.append(item)
+        # Return the list of matching items
+        return items_matching_category
```

Comment:
Notice this pattern of: 
```python
result_list = []
for element in source_list:
    if some_condition(element):
        result_list.append(element)
```

can be rewritten using a list comprehension as:
```python
result_list = [element for element in source_list if some_condition(element)]
```

Which here would look like:
```python
items_matching_category = [item for item in self.inventory if item.get_category() == category]
return items_matching_category
```

At first, this may seem more difficult to read, but comprehensions are a very common python style, so I encourage y’all to try working with them!